cluster/openstack-heat/util.sh

```diff
@@ -0,0 +1,274 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A library of helper functions that each provider hosting Kubernetes must implement to use cluster/kube-*.sh scripts.
+
+# exit on any error
+set -e
+
+# Use the config file specified in $KUBE_CONFIG_FILE, or default to
+# config-default.sh.
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+readonly ROOT=$(dirname "${BASH_SOURCE}")
+source "${ROOT}/${KUBE_CONFIG_FILE:-"config-default.sh"}"
+source "${KUBE_ROOT}/cluster/common.sh"
+if [ $CREATE_IMAGE = true ]; then
+source "${ROOT}/config-image.sh"
+fi
+
+# Verify prereqs on host machine
+function verify-prereqs() {
+ # Check the OpenStack command-line clients
+ for client in swift glance nova openstack;
```

Comment:
@idvoretskyi There's some sort of subcommand difference depending on what version of openstackclient you're using. @xsgordon is the one who ran into the bug, they'll have more details.

No matter what, we are going to have to prescribe a specific version of openstackclient to make sure this works. The individual commands(heat, swift, nova, etc) appear to have more stable arguments and flags. However there were concerns about them being deprecated, and indeed in the most recent versions of the individual commands a huge, distracting deprecation notice was displayed every time a command was issued.

There will always be restrictions for the version of the client, and the restrictions are listed in the documentation, and SHOULD in the future be checked by this provider.